### ISSUE TITLE
Nested override bug when using "Export Selected Artboards…"

### ISSUE THREAD

**Issue by tizzle** [andrewfiorillo/sketch-artboards-to-pdf#11]
Hey,

i'm using v1.1 and i experienced a bug with the "Export Selected Artboards…" function today. Normally i only use the "Export Current Page…" so i can't exactly say if this happens in simpler cases.

Unfortunately i cannot share my file, as it's under NDA.

I have an overlay with a close button (a blue 'x'). The button itself is a symbol, that itself is constructed of two more symbols, one for the background and one for the icon. The original button uses a different icon, an arrow pointing to the right.

To replace the original arrow with an X, i use nested overrides in my overlay. The exported PDF though does not show the X as expected, but the original symbol of the button, the arrow pointing to the right.

I hope you know what i mean... it's a bit hard to explain. I can try to provide a reduced test case that is not under NDA, if you need it.

Cheers! And keep up the great work, love your plugin!

Till

**Comment by davigoo** [comment]
Firstly thanks for building this useful plugin. I have experienced the same problem as above. In laymen, I would describe the issue the following way; Symbols are exported to PDFs even if they are set as hidden. I use a grid as a symbol and when I export, the grid is exported too, even though it's hidden.

**Comment by andrewfiorillo** [comment]
I'm having trouble reproducing this issue. Could either of you provide me with a test file, where this happens?

**Comment by tizzle** [comment]
I will try to prepare a file, i updated sketch yesterday, so i have to see if this can be reproduced as well-

**Comment by tizzle** [comment]
Here is a simplified reproduction. The problem occurs with the overridden icons in the buttons in the right column. See the sketch file and compare it to the exported PDF. Let me know if you need more details.

Thanks for looking into this!

[ARTBOARD_TO_PDF.zip](https://github.com/andrewfiorillo/sketch-artboards-to-pdf/files/910711/ARTBOARD_TO_PDF.zip)

**Comment by davigoo** [comment]
I have replicated the issue, see steps and attachments below.

Step 1:
I have 3 artboards all sharing the same symbol

Step 2: I hide the symbols as I don't want it to be included in the exported PDF. 

Step 3: I select board 1 and 3 in the layer list

Step 4: From the main menu I select Plugins> Artboards to PDF> Export Selected Artboards. This is to ensure that artboard 2 is excluded in the PDF export.

Step 5: The PDF includes the grid symbol even though it was hidden. 
Snapshot
https://www.evernote.com/l/ABKivR3FLGpOI5Fe15UCPMSU9KKiP-8ZFhAB/image.png

Sketch file:
[PDF Export.zip](https://github.com/andrewfiorillo/sketch-artboards-to-pdf/files/910981/PDF.Export.zip)

Hope this helps.

**Comment by andrewfiorillo** [comment]
@davigoo  ok, I think I've fixed your issue. Can you download this fork and see if that works for you:
https://github.com/andrewfiorillo/sketch-artboards-to-pdf/tree/hidden-symbol-fix

Thanks!

**Comment by davigoo** [comment]
@andrewfiorillo Just tested it and the hidden symbol stays hidden, great! Thanks for resolving so quickly.

**Comment by andrewfiorillo** [comment]
@davigoo I just merged the fixed into master. Thanks for providing the file and steps to reproduce.

@tizzle I am able to reproduce your issue, but I don't believe it is caused by the plugin. It appears to be a bug with in Sketch. This plugin relies on detaching symbols to avoid visual bugs in the output. But if you manually detach nested symbols consecutively after a detaching a parent symbol, you'll see it reverts to the original state, before the override. Also, if you use Sketch's built-in PDF export in the File menu, you'll see it has the same problem you described. Anyway, I'm going to close this issue, since @davigoo's problem was solved, and the nested symbol problem doesn't seem to be solvable unless there is a change to Sketch itself.
